packages/legacy/core/App/modules/openid/context/OpenIDCredentialRecordProvider.tsx

```diff
@@ -0,0 +1,151 @@
+  w3cCredentialRecords: Array<W3cCredentialRecord>
+  sdJwtVcRecords: Array<SdJwtVcRecord>
+  isLoading: boolean
+}
+
+const addRecord = (record: W3cCredentialRecord, state: OpenIDCredentialRecordState): OpenIDCredentialRecordState => {
+  const newRecordsState = [...state.w3cCredentialRecords]
+  newRecordsState.unshift(record)
+
+  return {
+    ...state,
+    w3cCredentialRecords: newRecordsState,
+  }
+}
+
+const removeRecord = (record: W3cCredentialRecord, state: OpenIDCredentialRecordState): OpenIDCredentialRecordState => {
+  const newRecordsState = [...state.w3cCredentialRecords]
+  const index = newRecordsState.findIndex((r) => r.id === record.id)
+  if (index > -1) {
+    newRecordsState.splice(index, 1)
+  }
+
+  return {
+    ...state,
+    w3cCredentialRecords: newRecordsState,
+  }
+}
+
+const defaultSate: OpenIDCredentialRecordState = {
+  openIDCredentialRecords: [],
+  w3cCredentialRecords: [],
+  sdJwtVcRecords: [],
+  isLoading: true,
+}
+
+interface Props {
+  children: React.ReactNode
+}
+
+const OpenIDCredentialRecordContext = createContext<OpenIDCredentialContext>(null as unknown as OpenIDCredentialContext)
+
+const isW3CCredentialRecord = (record: W3cCredentialRecord) => {
+  return record.getTags()?.claimFormat === ClaimFormat.JwtVc
+}
+
+const filterW3CCredentialsOnly = (credentials: W3cCredentialRecord[]) => {
+  return credentials.filter((r) => isW3CCredentialRecord(r))
+}
+
+// eslint-disable-next-line react/prop-types
+export const OpenIDCredentialRecordProvider: React.FC<PropsWithChildren<Props>> = ({ children }) => {
+  const [state, setState] = useState<OpenIDCredentialRecordState>(defaultSate)
+
+  const { agent } = useAgent()
+  const [logger] = useServices([TOKENS.UTIL_LOGGER])
+
+  function checkAgent() {
+    if (!agent) {
+      const error = 'Agent undefined!'
+      logger.error(`[OpenIDCredentialRecordProvider][deleteCredential] ${error}`)
+      throw new Error(error)
+    }
+  }
+
+  async function storeCredential(cred: W3cCredentialRecord | SdJwtVcRecord): Promise<void> {
+    checkAgent()
+    if (cred instanceof W3cCredentialRecord) {
+      await agent?.dependencyManager.resolve(W3cCredentialRepository).save(agent.context, cred)
+    } else {
+      await agent?.dependencyManager.resolve(SdJwtVcRepository).save(agent.context, cred)
+    }
+  }
+
+  async function deleteCredential(cred: W3cCredentialRecord | SdJwtVcRecord) {
+    checkAgent()
+    if (cred instanceof W3cCredentialRecord) {
+      await agent?.w3cCredentials.removeCredentialRecord(cred.id)
+    } else if (cred instanceof SdJwtVcRecord) {
+      await agent?.sdJwtVc.deleteById(cred.id)
+    }
+  }
+
+  useEffect(() => {
+    if (!agent) {
+      return
+    }
+    agent.w3cCredentials?.getAllCredentialRecords().then((w3cCredentialRecords) => {
+      setState({ ...state, w3cCredentialRecords: filterW3CCredentialsOnly(w3cCredentialRecords), isLoading: false })
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Do you intend to remove this line and fix the lint error later because it's still a work in progress? If so, that's ok. Otherwise we should probably fix the lint error now.

Reply:
This is one of the particular cases where exhaustive deps will need much refactoring to satisfy
If you notice, the linter require adding state as dependency, if you do that, you cause an infinite loop, because we resetting the state inside the useEffect. If you have better refactor please let me know

Reply:
```suggestion
    setState((prev) => ({ ...prev, w3cCredentialRecords: filterW3CCredentialsOnly(w3cCredentialRecords), isLoading: false }))
```

Could this work? Then `state` doesn't have to be a dep.

Reply:
@MosCD3 , can you use the react [updater function](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state) approach to avoid disabling the linter?